### PR TITLE
add 21.4 tests (SC-609)

### PR DESCRIPTION
**Pull request by holmanb**
```
add miscellaneous tests

Add growpart integration test and a few unittests.
```

## Test Steps
set PLATFORM = 'lxd_vm' for the integration test

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: tests/integration_tests/modules/test_growpart.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@pytest.mark.ubuntu
+@pytest.mark.lxd_vm
+class TestGrowPart:
+    """Test growpart"""
+
+    def test_grow_part(self, client: IntegrationInstance):
+        """Verify """
+        log = client.read_from_file('/var/log/cloud-init.log')
+        assert ("cc_growpart.py[INFO]: '/dev/sdb1' resized:"
+                " changed (/dev/sdb, 1) from") in log
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_growpart.py:55]: I think it's worth an additional assertion that the disk was actually resized correctly using something like `lsblk`. Similar to https://github.com/canonical/cloud-init/blob/main/tests/integration_tests/modules/test_disk_setup.py#L171

File: tests/unittests/test_handler/test_handler_growpart.py [hunk 2]
```diff
@@ -130,7 +162,42 @@ def test_mode_auto_prefers_growpart(self):
...
+    @mock.patch.object(os, 'scandir', return_value=Scanner())
+    @mock.patch.object(os, 'mkdir')
+    @mock.patch.object(os, 'unlink')
+    @mock.patch.object(os, 'rmdir')
+    @mock.patch.object(os, 'open', return_value=1)
+    @mock.patch.object(os, 'close')
+    @mock.patch.object(shutil, 'rmtree')
+    @mock.patch.object(os, 'lseek', return_value=1024)
+    @mock.patch.object(os, 'lstat', return_value='interesting metadata')
+    def test_force_lang_check_tempfile(self, *args, **kwargs):
```
Comment (TheRealFalcon) [inline tests/unittests/test_handler/test_handler_growpart.py:179]: Ooph, that's a lot of mocks. Not bad...but some of the base classes we don't use as often make a little more sense now :wink:

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Nice test additions/changes. Thanks!

My only request is an additional check on the integration that we actually did what we said we did in the log.

File: tests/unittests/test_handler/test_handler_growpart.py (hunk 2 above)
Comment (holmanb) [reply tests/unittests/test_handler/test_handler_growpart.py:179]: I figured you might say that :). It was a good exercise in understanding "why" those test classes exist.

**Comment by holmanb** [comment]
I resolved the merge conflict and addressed the requests. I think this is ready for review / merge.

**Review by TheRealFalcon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**